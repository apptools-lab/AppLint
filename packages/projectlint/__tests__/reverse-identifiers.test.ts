import { describe } from 'vitest';

const { defineTest } = require('jscodeshift/dist/testUtils');
const { defineInlineTest } = require('jscodeshift/dist/testUtils');
const { defineSnapshotTest } = require('jscodeshift/dist/testUtils');
const transform = require('../reverse-identifiers');

defineTest(__dirname, 'reverse-identifiers');

// defineTest(__dirname, 'reverse-identifiers', null, 'typescript/reverse-identifiers', { parser: 'ts' });

describe('reverse-identifiers', () => {
  defineInlineTest(transform, {}, `
var firstWord = 'Hello ';
var secondWord = 'world';
var message = firstWord + secondWord;`, `
var droWtsrif = 'Hello ';
var droWdnoces = 'world';
var egassem = droWtsrif + droWdnoces;
  `);
  defineInlineTest(transform, {},
    'function aFunction() {};',
    'function noitcnuFa() {};',
    'Reverses function names',
  );
});

// the snapshot output of this file should be the same as reverse-identifiers.output.js
defineSnapshotTest(transform, {}, 'reverse-identifiers');