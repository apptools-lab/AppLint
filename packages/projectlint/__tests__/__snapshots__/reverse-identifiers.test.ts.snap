// Vitest Snapshot v1

exports[`transforms correctly 1`] = `"esrever-sreifitnedi"`;
